library/common/extensions/filters/http/platform_bridge/filter.cc

```diff
@@ -325,7 +369,20 @@ PlatformBridgeFilter::encodeTrailers(Http::ResponseTrailerMap& trailers) {
   return status;
 }
 
+void PlatformBridgeFilter::resumeDecoding() {
+  auto weak_self = weak_from_this();
+  dispatcher_.post([weak_self]() -> void {
```

Comment:
I want to call out that there is a potential shutdown race here, but it all depends on timing. We don't retain the filter and so when Envoy is being torn down, it's possible the filter is torn down before the dispatcher, in which case everything is fine. If the inverse happens, however, then there's narrow window here where we could successfully acquire the filter and attempt to access the dispatcher, which is already gone.

As we discussed in person, there are a few solutions to this. We could allow the dispatcher itself to be shared and locked (this is seems ideal to me, since we would never have a garbage reference here or anywhere else we needed to call in from platform code on an arbitrary thread). We could lock the top-level engine object, but the downside is that all incoming calls we need access to it (and indeed, there's not an obvious reason a filter should have access to it, let alone a mechanism to make it available short of ugly hacks). And finally, we could take ownership of the dispatcher and have it be something passed into Envoy to ensure it was always valid. (There are other advantages to this last approach, and it might be desirable eventually regardless, but it would initially require non-trivial changes to Envoy.)

Reply:
can we leave a todo in the code linking to an issue that we can track for any of the solutions you list?

Reply:
yes